### PR TITLE
[image_picker] Migrate to null-safety

### PR DESCRIPTION
**Pull request by Sameerkash**
Migrate `image_picker` to null-safety

This PR should close flutter/flutter#75162

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by Sameerkash** [comment]
@cyanglaz Looks like the example needs `image_picker_web` to be migrated. If it's still no WIP on that, I'll update that and push it along with this.

File: packages/image_picker/image_picker/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+## 0.7.0-nullsafety
+* Migrate to nullsafety
+* Breaking Changes:
+    * Removed the deprecated methods: `ImagePicker.pickImage`, `ImagePicker.pickVideo`,
+`ImagePicker.retrieveLostData`
+
+* iOS: update XCUITests to separate each test session.
```
Comment (stuartmorgan) [inline packages/image_picker/image_picker/CHANGELOG.md:7]: This looks like it was accidentally duplicated from the previous notes.

File: packages/image_picker/image_picker/pubspec.yaml [hunk 2]
```diff
@@ -16,19 +16,19 @@ flutter:
...
   integration_test:
     path: ../../integration_test
-  mockito: ^4.1.3
+  mockito: ^5.0.0-nullsafety.7
   pedantic: ^1.8.0
   plugin_platform_interface: ^1.0.3
 
 environment:
-  sdk: ">=2.1.0 <3.0.0"
+  sdk: '>=2.12.0-0 <3.0.0'
```
Comment (stuartmorgan) [inline packages/image_picker/image_picker/pubspec.yaml:33]: Let's keep the use of " rather than ' consistent here.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Looks good generally. You can remove image_picker_web from the example for now, since it's currently unendorsed anyway, to resolve the issues with dependency conflicts in the example.

File: packages/image_picker/image_picker/CHANGELOG.md (hunk 1 above)
Comment (Sameerkash) [reply packages/image_picker/image_picker/CHANGELOG.md:7]: How do you suggest I put it?

Comment (stuartmorgan) [reply packages/image_picker/image_picker/CHANGELOG.md:7]: I'm not sure what you're asking. The solution to having accidentally copied and pasted a line from the previous version's release notes is to remove the accidental duplicate.

Comment (Sameerkash) [reply packages/image_picker/image_picker/CHANGELOG.md:7]: Ahh!, I'm sorry I missed that line. I thought you were talking about the way I put the breaking changes and thought some wording needed to be changed. I'll fix that.

**Review by stuartmorgan — APPROVED** [review]
LGTM! (I did a little bit of minor final cleanup)

**Comment by jianhuaz** [comment]
Can a provide a version, modify the flitter_ plugin_ android_ Version,  to support flutter 1.22.

**Pull request merged**